tests/test_dynamodb/exceptions/test_dynamodb_exceptions.py

```diff
@@ -761,3 +761,42 @@ def test_query_begins_with_without_brackets():
         'Invalid KeyConditionExpression: Syntax error; token: "sk"'
     )
     err["Code"].should.equal("ValidationException")
+
+
+@mock_dynamodb
+def test_transact_write_items_multiple_operations_fail():
+
+    # Setup
+    table_schema = {
+        "KeySchema": [{"AttributeName": "id", "KeyType": "HASH"}],
+        "AttributeDefinitions": [{"AttributeName": "id", "AttributeType": "S"}],
+    }
+    dynamodb = boto3.client("dynamodb", region_name="us-east-1")
+    table_name = "test-table"
+    dynamodb.create_table(
+        TableName=table_name, BillingMode="PAY_PER_REQUEST", **table_schema
+    )
+
+    # Execute
+    with pytest.raises(ClientError) as exc:
+        dynamodb.transact_write_items(
+            TransactItems=[
+                {
+                    "Put": {
+                        "Item": {"id": {"S": "test"}},
+                        "TableName": table_name,
+                    },
+                    "Delete": {
+                        "Key": {"id": {"S": "test"}},
+                        "TableName": table_name,
+                    },
+                }
+            ]
+        )
+        # Verify
+        err = exc.value.response["Error"]
+        assert err["Code"] == "ValidationException"
+        assert (
+            err["Message"]
+            == "TransactItems can only contain one of Check, Put, Update or Delete"
+        )
```

Comment:
```suggestion
    # Verify
    err = exc.value.response["Error"]
    assert err["Code"] == "ValidationException"
    assert (
        err["Message"]
        == "TransactItems can only contain one of Check, Put, Update or Delete"
)
```

As it stands, if `transact_write_items` throws an exception, the verification code is never run.

Reply:
Good catch